test/screenshot/lib/controller.js

```diff
@@ -229,77 +250,62 @@ class Controller {
    * @return {!Promise<!Buffer>}
    * @private
    */
-  async downloadImage_(uri) {
-    // For binary data, set `encoding: null` to return the response body as a `Buffer` instead of a `string`.
-    // https://github.com/request/request#requestoptions-callback
-    return request({uri, encoding: null})
-      .then(
-        (body) => this.autoCropImage_(body),
-        (err) => {
-          console.error(`FAILED to download "${uri}"`);
-          return Promise.reject(err);
-        }
-      );
-  }
-
-  /**
-   * @param {!Buffer} imageData Uncropped image buffer
-   * @return {!Promise<!Buffer>} Cropped image buffer
-   * @private
-   */
-  async autoCropImage_(imageData) {
-    return jimp.read(imageData)
-      .then(
-        (image) => {
-          return new Promise((resolve, reject) => {
-            image
-              .autocrop()
-              .getBuffer(jimp.MIME_PNG, (err, buffer) => {
-                return err ? reject(err) : resolve(buffer);
-              });
-          });
-        },
-        (err) => Promise.reject(err)
-      );
+  async downloadAndCropImage_(uri) {
+    const uncroppedImageData = await this.imageCache_.getImageBuffer(uri);
+    const croppedImageData = this.imageCropper_.autoCropImage(uncroppedImageData);
```

Comment:
The "no awaiting the final promise" pattern is weirding me out. I'm not saying you should change it but it feels unnatural.

Reply:
Woops! That is indeed very confusing. I suppose I could just inline both variables:

```js
  async downloadAndCropImage_(uri) {
    return this.imageCropper_.autoCropImage(await this.imageCache_.getImageBuffer(uri));
  }
```